### PR TITLE
feat: trkrclusterv6 implementation

### PR DESCRIPTION
**Pull request by osbornjd**
This implements the v6 option now without any performance changes

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

## Links to other PRs in macros and calibration repositories (if applicable)

AI-assisted summary; AI can make mistakes, so please use best judgment when reviewing.

Motivation / context
- Add support for `trkrclusterv6` on `master` while keeping the existing v5 clustering path intact.
- Enrich TPC cluster diagnostics/ntupling to expose additional overlap/edge/mix and geometry-related metadata (including event identifiers) for improved performance studies and validation.

Key changes
- TPC clustering (`TpcClusterizer`):
  - Introduce a richer `ClusterCounters`-based flow to classify overlap plus detailed edge/mixing categories (phi/time edges, dead/hot edges, mix flags) and drive cluster parameter computation.
  - Add a debug-gated selection between `TrkrClusterv5` and `TrkrClusterv6` when constructing cluster objects.
  - Propagate the global `m_debug` into threaded clustering context.
  - Add `DetailedClusterAnalysis()` to enable extended debug behavior.
- Track/cluster data model (`TrkrCluster` and cluster implementations):
  - Update `TrkrCluster` virtual setter/getter signatures to tighten const-correctness and align return types (e.g., `getPadMax`/`getTBinMax` and several mix/bin/phase getters) with the v6/diagnostic needs.
  - Update `TrkrClusterv5`/`TrkrClusterv6` method signatures to match the tightened interface expectations.
- Tracking diagnostics + ntupling:
  - `TrackResiduals`: capture `evt_id` from the event header and write it into multiple ROOT trees/branches; expand cluster-tree/residual-tree branch schema with additional cluster metadata (ADC center/max, geometry-derived size/bin/phase fields, detailed edge/mix/overlap counts and bounds).
  - ROOT branch leaf type updates: cluster edge/mix indicators are written with `"/B"` leaf specifiers (e.g., `m_overlap/B`, `m_slmix/B`, etc.), reflecting the intended small-integer/char storage.
  - `TrkrNtuplizer`: extend NTuple cluster column list and enum indices; fill new cluster fields from additional `TrkrCluster` getters; adjust which cluster “size” getter is used.

Potential risk areas
- IO/schema compatibility: ROOT tree branch/leaf lists and ntuple column sets/types are extended; downstream analysis reading these outputs may require updates.
- Reconstruction/physics behavior: v6 cluster construction and the expanded edge/overlap/mix classification logic could change derived cluster properties used downstream.
- API changes: `TrkrCluster`/`TrkrClusterv5`/`TrkrClusterv6` signature and return-type changes may affect any external code building against these classes.
- Threading/debug behavior: v6 selection and detailed analysis are tied to debug/thread data plumbing; validate behavior in multi-threaded runs.
- Performance: clustering logic is refactored (even if intended to be performance-neutral); confirm with representative workloads.

Possible future improvements
- Add explicit configuration/QA checks that confirm v5 vs v6 behavior equivalence where expected (and quantify differences where classification is intentionally richer).
- Provide documentation/migration notes for downstream consumers of the expanded ROOT trees/ntuples (new fields, types, and semantics).
- Add regression tests for edge/mix/overlap categorization and for schema stability across releases.

### CODE REVIEW

File: offline/packages/tpc/TpcClusterizer.cc [hunk 1]
```diff
@@ -176,37 +209,44 @@ namespace
...
-        edge++;
+        counts.nedge++;
       }
       if ((tbin - tdown) <= 0)
       {
         tdown = tbin;
-        edge++;
+        counts.nedge++;
       }
       return;
```
[the remark this answers is not in the document]
Comment (osbornjd) [reply offline/packages/tpc/TpcClusterizer.cc:234]: This will be fixed in a future PR
